### PR TITLE
initializersRegExp updated to handle instance initializers

### LINKED ISSUE #13: Work needed here to support instance initializers
Just opening this here since the discussion started in ember-cli:
https://github.com/ember-cli/ember-cli/issues/3159

I'd be happy to take this on, even if it's a bit premature :)

### PR DESCRIPTION
**Pull request by jasonmit**
fixes #13

### CODE REVIEW

**Comment by jasonmit** [comment]
Ignore test failure, there aren't tests for this.

**Comment by rwjblue** [comment]
Would need call `app.instanceInitializer` for the instance initializers (not `initializer` as it is now).

Tests in Ember showing this:  https://github.com/emberjs/ember.js/blob/06e41ad7ccd28558dd4e651aa070bc06f7757821/packages/ember-application/tests/system/instance_initializers_test.js#L45-L50

**Comment by jasonmit** [comment]
@rwjblue sorry about that.  Fixed.

File: ember-load-initializers.js [hunk 1]
```diff
@@ -6,22 +6,32 @@ define("ember/load-initializers",
...
-          if (!initializer.name) {
-            var initializerName = moduleName.match(/[^\/]+\/?$/)[0];
-            initializer.name = initializerName;
-          }
+            var module = require(moduleName, null, null, true);
+            if (!module) { throw new Error(moduleName + ' must export an initializer.'); }
 
-          app.initializer(initializer);
-        });
+            var initializerType = Ember.String.camelize(dep.matches[1].substring(0, dep.matches[1].length - 1));
```
Comment (jasonmit) [inline ember-load-initializers.js]:
```
initializers => initializer
instance-initializers => instanceInitializer
```

**Comment by rwjblue** [comment]
LGTM

**Comment by rwjblue** [comment]
Tagged as v0.1.0.

@jasonmit - Can you update Ember CLI itself to use the newer version?

**Comment by jasonmit** [comment]
:+1: will do, thanks

**Comment by stefanpenner** [comment]
thanks @jasonmit

**Comment by atomkirk** [comment]
What about a generator for instance initializers?

**Comment by brandonparsons** [comment]
@rwjblue @stefanpenner  Does this patch exist in `ember-cli 0.2.7`? I've tried putting my initializer into `/app/instance-initializers` but it still seems to think that I have a regular (i.e. I can't call `lookup()` on the first argument).

**Comment by givanse** [comment]
@atomkirk https://github.com/ember-cli/ember-cli/issues/4198

**Pull request merged**